addons/misra.py

```diff
@@ -255,6 +255,36 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand1,
             lpar.astOperand2.astOperand2.astOperand2]
 
+def getWhileLoopExpressions(whileToken):
+    if not whileToken or whileToken.str != 'while':
+        return None
+    lpar = whileToken.next
+    if not lpar or lpar.str != '(':
+        return None
+    if not lpar.astOperand2.astOperand1:
+        return None
+    counter = lpar.astOperand2.astOperand1.str
+    token = lpar
+    while (token and token.str != "{"):
```

Comment:
use the token.link:

    rpar = lpar.link

and then this check might be appropriate

    if not simpleMatch(rpar, ') {'):
        return None